### PR TITLE
query: Added experimental federate endpoint.

### PR DESCRIPTION
**Pull request by magiceses**
This adds /federate endpoint directly to Querier.

It is a example.
image

### CODE REVIEW

**Review by hoperays — COMMENTED** [review]
LGTM

File: pkg/query/adapter/adapter.go [hunk 1]
```diff
@@ -0,0 +1,494 @@
+package adapter
```
Comment (hoperays) [inline pkg/query/adapter/adapter.go:4]: Please add the same license header as cmd/thanos/query.go

File: pkg/query/adapter/adapter.go [hunk 2]
```diff
@@ -0,0 +1,494 @@
...
+		if err != nil {
+			return nil, &api.ApiError{Typ: api.ErrorBadData, Err: err}
+		}
+		storeMatchers = append(storeMatchers, matchers)
+	}
+
+	return storeMatchers, nil
+}
+
+func (ad *PrometheusAdapter) parseLookbackDeltaParam(r *http.Request) (time.Duration, *api.ApiError) {
```
Comment (hoperays) [inline pkg/query/adapter/adapter.go]: not used?

**Review by hoperays — CHANGES_REQUESTED** [review]

File: pkg/query/adapter/adapter.go (hunk 1 above)
Comment (magiceses) [reply pkg/query/adapter/adapter.go:4]: done

File: pkg/query/adapter/adapter.go (hunk 2 above)
Comment (magiceses) [reply pkg/query/adapter/adapter.go]: deleted. done.

**Review by hoperays — APPROVED** [review]

**Pull request merged**